common/djangoapps/student/views.py

```diff
@@ -2099,6 +2101,29 @@ def create_account_with_params(request, params):
     return new_user
 
 
+def _apply_third_party_auth_overrides(request, extra_fields):
+    """
+    Modify the extra_fields and required attributes of the corresponding fields if the user has
+    authenticated with a third-party provider.
+    """
+    updated_fields = deepcopy(extra_fields)
+
+    if third_party_auth.is_enabled():
+        running_pipeline = third_party_auth.pipeline.get(request)
+        if running_pipeline:
+            current_provider = third_party_auth.provider.Registry.get_from_pipeline(running_pipeline)
+            # Make input fields read-only for fields passed in via SSO callback data.
+            if current_provider.sync_learner_profile_data:
+                display_field_names = ['terms_of_service', 'honor_code', 'email', 'name']
```

Comment:
It would be great if we can set this value from settings as we are using same list in user API too.